### ISSUE TITLE
Missing python dependencies

### ISSUE THREAD

**Issue by GMishx** [astraw/stdeb#138]
I have created my project which have both the `setup.py` and `requirements.txt`.

I have listed all the dependencies along with their versions in `setup.py`'s `setup_requires` and `install_requires`. These dependencies gets installed when I run `python setup.py install`.

But I created the `.deb` file using `python3 setup.py --command-packages=stdeb.command bdist_deb` and copied the `.deb` file on a new system and installed the deb using `dpkg -i <package>.deb`. It had almost all the dependencies installed using `apt-get install -f` instead of `pip install`.

This resulted in some missing dependencies and I got this error:
```
Traceback (most recent call last):
  File "/usr/bin/mypackage", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2927, in <module>
    @_call_aside
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2913, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2940, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 635, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 943, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 829, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'textdistance>=3.0.3' distribution was not found and is required by mypackage
```

Any idea how can I get these dependencies installed by pip when installing the deb file on a new system?

**Comment by dirk-thomas** [comment]
> Any idea how can I get these dependencies installed by pip when installing the deb file on a new system?

A Debian package can only depend on other Debian packages. So it can't install dependencies using `pip`.

> `The 'textdistance>=3.0.3' distribution was not found and is required by mypackage`

I assume there is no Debian package of `textdistance`?

If you want install the dependencies from `pip` why not install the package in question from `pip` too?

Assuming my comment answered your question I will go ahead and close this ticket for now. Please feel free to continue commenting and the ticket can be reopened if necessary.

**Comment by mendhak** [comment]
> If you want install the dependencies from pip why not install the package in question from pip too?

Hey @dirk-thomas  could you elaborate on this bit, I'm also quite new to this.  Suppose some package xyz in `install_requires` has no debian equivalent - and the only way to get it is from pip.  How could we ensure that the package xyz is on the user's system so that our library/application/script can run?

**Comment by dirk-thomas** [comment]
> Suppose some package xyz in `install_requires` has no debian equivalent - and the only way to get it is from pip.  How could we ensure that the package xyz is on the user's system so that our library/application/script can run? 

@mendhak You package needs to declare its requirements in the `setup.py/cfg` file using the `install_requires` argument so when you install your package via `pip` it will automatically install those dependencies if not already present (that also works recursively). Please check the Python documentation for more information since this is not related to `stdeb` in any way.

**Comment by mendhak** [comment]
Oh I wasn't clear, sorry let me try again.  
  
Suppose we're distributing a Python application via `.deb`  (like a Qt or GTK application)
But there are some dependencies needed, which are only available via pip.  
What's an approach we can take there?  
I believe this is the same question that OP is asking.

**Comment by dirk-thomas** [comment]
I think I have already answered that question above: a Debian packages can only depend on other Debian packages - not on `pip` packages. So either you also release the dependencies as Debian packages or you must install all from `pip`.

**Comment by oren0e** [comment]
@dirk-thomas: I have a similar problem, I created the deb versions of the dependencies and tired to specify them in `stdeb.cfg` with `Depends: python3-dep1` - but still when I build the package with `python3 setup.py --command-packages=stdeb.command bdist_deb` the output prints it can't find these, messages like:
```
I: dh_python3 pydist:224: Cannot find package that provides dep1. Please add package that provides it to Build-Depends or add "dep1 python3-dep1" line to debian/py3dist-overrides or add proper dependency to Depends by hand and ignore this info.
```
What should I do? I want the dependencies to be installed as well automatically

**Comment by dirk-thomas** [comment]
@oren0e Sorry, I am not a maintainer of this repository anymore.
